containers/User/SettingsProfile.js

```diff
@@ -1,24 +1,120 @@
-import React from 'react'
-import { View } from 'react-native'
+import React, { useState, useEffect } from 'react'
+import { Alert, View } from 'react-native'
+import { Formik } from 'formik'
+import * as yup from 'yup'
+import AsyncStorage from '@react-native-async-storage/async-storage'
+import API from '../../BaseApi'
 import { ButtonSubmit } from '../../components/ButtonSubmit'
 import { Header } from '../../components/Header'
-import { TextSourceSans } from '../../components/TextSourceSans'
 import { TextInputFormField } from '../../components/formFields'
+import { ListContainer, ListItem } from '../../components/List'
+import { AvatarCircle } from '../../components/ButtonAvatar'
+import { styles } from './SettingsProfile.styles'
 
 export const SettingsProfile = props => {
 
+  const {userId} = props.route.params
+  const [user, setUser] = useState()
+
+  const userNameValidationSchema = yup.object().shape({
+    username: yup
+      .string()
+      .max(60, 'Required. 60 characters or fewer. Letters, digits, spaces and @/./+/-/_ only.')
+  })
+
+  const handleSubmit = (values) => {
+    const username = values.username
+    const userimage = user._avatar
+    AsyncStorage.getItem('authToken')
+      .then((token) => {
+        return API.editUser({username, userimage}, token)
+      })
+      .then((response) => {
+        const {statusCode, data} = response
+        if (statusCode == 200) {
+          Alert.alert('You\'re Username has been updated')
+          props.navigation.goBack()
+        } else {
+          const errorMessage = 'That did not work.'
+          let errorDetail
+          if (statusCode==403) {
+            errorDetail = data.detail
+          } else if (statusCode == 400) {
+            errorDetail = 'Bad request'
+          }
+          Alert.alert(errorMessage, errorDetail)
+        }
+      })
+  }
+
+  const fetchUser = () => {
+    return AsyncStorage.getItem('authToken')
+      .then((token) => API.getUser(userId, token))
+      .then((response) => {
+        if(response.statusCode === 200) {
+          setUser(response.data)
+        } else {
+          return Promise.reject(new Error('fetchUser returned ' + response.statusCode))
+        }
+      })
+      .catch(error => console.warn(error))
+  }
+
```

Comment:
@Rineee as I need to update the input placeholder I re-fetched the user here but is there a more efficient way to do it? i didn't think so but wanted to check

Reply:
I think the update should also give you back the user, so you could just use the response.date from edit response and setUser to that data, would that work?

Reply:
Ah, I think I got it wrong, you maybe meant initially when you get on settings page? Cant we pass the the data there?

Reply:
yep, I was passing the data but then `ProfileSettings` data wasn't updating after save as it was still using the data passed from `Profile`
But you're saying to pass data for initial render and get updated data from response, however i'm not sure how to tell it which to use, as for initial rendering it will use the data from 'Profile' again, does that make sense?